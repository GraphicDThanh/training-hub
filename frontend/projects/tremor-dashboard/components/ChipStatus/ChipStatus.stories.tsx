// Libs
import type { Meta, StoryObj } from "@storybook/react";

// Components
import ChipStatus from "./ChipStatus";

const meta = {
  title: "Components/Common/ChipStatus",
  tags: ["autodocs"],
  component: ChipStatus,
  argTypes: {
    extendedClass: {
      description: "Class of button",
    },
    value: {
      description: "Value of button",
    },
  },
} as Meta<typeof ChipStatus>;

export default meta;

type Story = StoryObj<typeof meta>;

export const ButtonIncrease: Story = {
  args: {
    extendedClass: "text-few border-few",
    value: "+ 1234",
  },
};

export const ButtonDecrease: Story = {
  args: {
    extendedClass: "text-attention border-attention",
    value: "- 1234",
  },
};

export const ButtonPending: Story = {
  args: {
    extendedClass: "text-primary border-primary",
    value: "Pending",
  },
};
