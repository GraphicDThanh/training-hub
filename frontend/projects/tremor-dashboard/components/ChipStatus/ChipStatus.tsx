import { BiChevronDown, BiChevronUp } from "react-icons/bi";
import { BsExclamationLg } from "react-icons/bs";

// Components
import { Flex } from "@tremor/react";
import { memo, useCallback } from "react";

interface StatusButtonProps {
  extendedClass: string;
  value: string;
}

const ChipStatus = ({ extendedClass, value }: StatusButtonProps) => {
  const renderIcon = useCallback(() => {
    if (value === "Pending") return <BsExclamationLg />;
    if (value.includes("+")) return <BiChevronUp />;

    return <BiChevronDown />;
  }, [value]);

  return (
    <Flex
      justifyContent="center"
      className={`shrink-0 w-9 h-9 border rounded-full ${extendedClass}`}>
      {renderIcon()}
    </Flex>
  );
};

export default memo(ChipStatus);
