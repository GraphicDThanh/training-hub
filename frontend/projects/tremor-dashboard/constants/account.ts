// Types
import { OptionType } from "@/types";

export const GENDERS: OptionType[] = [
  {
    option: "Male",
    value: "0",
  },
  {
    option: "Female",
    value: "1",
  },
];

export const SKILLS: Record<string, string | number>[] = [
  {
    value: "0",
    option: "React",
  },
  {
    value: "1",
    option: "Vue",
  },
  {
    value: "2",
    option: "Angular",
  },
  {
    value: "3",
    option: "Svelte",
  },
  {
    value: "4",
    option: "Javascript",
  },
];

export const MONTHS: OptionType[] = [
  {
    option: "January",
    value: "0",
  },
  {
    option: "February",
    value: "1",
  },
  {
    option: "March",
    value: "2",
  },
  {
    option: "April",
    value: "3",
  },
  {
    option: "May",
    value: "4",
  },
  {
    option: "June",
    value: "5",
  },
  {
    option: "July",
    value: "6",
  },
  {
    option: "August",
    value: "7",
  },
  {
    option: "September",
    value: "8",
  },
  {
    option: "October",
    value: "9",
  },
  {
    option: "November",
    value: "10",
  },
  {
    option: "December",
    value: "11",
  },
];

export const ROLES = [
  {
    value: 0,
    label: "User",
  },
  {
    value: 1,
    label: "Admin",
  },
];
