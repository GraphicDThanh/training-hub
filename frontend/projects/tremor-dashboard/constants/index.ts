export * from "./account";
export * from "./analytics";
export * from "./api";
export * from "./billing";
export * from "./breakpoints";
export * from "./colors";
export * from "./common";
export * from "./flagSrc";
export * from "./images";
export * from "./pagination";
export * from "./messages";
export * from "./pincode";
export * from "./products";
export * from "./projects";
export * from "./profile";
export * from "./regex";
export * from "./routes";
export * from "./sales";
export * from "./saleStatistical";
export * from "./sections";
export * from "./status";
export * from "./units";
export * from "./keyboard";
export * from "./steps";
export * from "./users";
export * from "./conversation";
