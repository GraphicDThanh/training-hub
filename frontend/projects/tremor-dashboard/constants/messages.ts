export const MESSAGES_ERROR = {
  NAME_MIN_LENGTH: "Minimum length should be 4.",
  NAME_INVALID: "This field is no space at start or end",
  NAME_REQUIRED: "Name is required.",
  PASSWORD_REQUIRED: "Password is required.",
  EMAIL_REQUIRED: "Email is required.",
  EMAIL_INVALID: "Invalid email address.",
  SKU_INVALID: "Invalid SKU number.",
  PASSWORD_INVALID:
    "Password must have at least 8 characters, 1 special character and 1 number.",
  FIELD_REQUIRED: "This field is required.",
  MIN_LENGTH_4: "This field must have more than 4 characters.",
  URL_REQUIRED: "URL is required.",
  INVALID_URL: "URL is invalid.",
  INVALID_PHONE_NUMBER: "Phone number is invalid",
  INVALID_BIRTH_DATE: "Birth date is invalid.",
  REPEAT_PASSWORD_INVALID: "Password not matching",
  TERMS_REQUIRED: "Please agree with the terms and conditions to continue.",
  CONFIRM_EMAIL: "The email confirmation does not match.",
  PRODUCT: {
    WEIGHT_LIMIT: "Weight has limit is 2000 pound.",
    QUANTITY_LIMIT: "Quantity has limit is 10000.",
    PRICE_LIMIT: "Price has limit is $20,000.",
  },
  TRANSACTION: {
    AMOUNT_LIMIT: "Amount has limit is $20,000.",
    MIN_AMOUNT: "Amount must be positive number.",
  },
};

export const MESSAGES_WARNING = {
  PRODUCT: {
    WEIGHT_IS_ZERO: "Missing weight is not encourage.",
    QUANTITY_IS_ZERO:
      "This product will has label sold out if have quantity default is 0.",
    ACCEPT_IMAGE_UPLOAD:
      "(We accept .png, .jpg, .jpeg, and .webp files, up to 32MB in size.)",
  },
};

export const MESSAGE_SIGN_UP = {
  SUCCESS: "Sign up successfully.",
  FAILED: "Failed to sign up",
};

export const MESSAGE_SIGN_IN = {
  FAILED: "Failed to sign in",
  SUCCESS: "Sign in successfully.",
  CREDENTIALS: "Please make sure your email address and password are correct.",
};

export const MESSAGE_ADD_PRODUCT = {
  SUCCESS: "Add product successfully",
  FAILED: "Failed to add product",
};

export const MESSAGE_EDIT_PRODUCT = {
  SUCCESS: "Edit product successfully",
  FAILED: "Failed to edit product",
};

export const MESSAGE_USER = {
  MAIL_EXISTS: "Account with this email already exists!",
  ADD_FAILED: "Failed to create new account. Please try again!",
  EDIT_SUCCESS: "Edit user successfully",
  EDIT_FAILED: "Failed to edit user",
  GET_USER_FAILED: "Failed to get user",
};

export const MESSAGE_SIGN_OUT = {
  SUCCESS: "Sign out successfully.",
  FAILED: "Failed to Sign out",
};

export const MESSAGE_ADD_PROJECT = {
  SUCCESS: "Add project successfully",
  FAILED: "Failed to add project",
};

export const MESSAGE_EDIT_PROJECT = {
  SUCCESS: "Edit project successfully",
  FAILED: "Failed to edit project",
};

export const MESSAGE_DELETE_PROJECT = {
  SUCCESS: "Delete project successfully",
  FAILED: "Failed to delete project",
};

export const MESSAGE_ADD_USER = {
  SUCCESS: "Add new user successfully",
  FAILED: "Failed to add new user",
};

export const MESSAGES_TRANSACTION = {
  ADD_SUCCESS: "Create new transaction successfully",
  ADD_FAILED: "Failed to create new transaction. Please try again!",
};
