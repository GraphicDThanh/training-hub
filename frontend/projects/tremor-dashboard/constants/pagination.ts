export const PAGE_SIZE = {
  SIZE: 10,
};
