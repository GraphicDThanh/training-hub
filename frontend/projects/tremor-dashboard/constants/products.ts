// Types
import { OptionType, TFieldSocial } from "@/types";

type TopSellingProductsType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
  5: string;
};

export const TOP_SELLING_PRODUCTS_IMAGE: TopSellingProductsType = {
  1: "/assets/images/products/blue-shoe.webp",
  2: "/assets/images/products/black-mug.webp",
  3: "/assets/images/products/black-chair.webp",
  4: "/assets/images/products/bang-sound.webp",
  5: "/assets/images/products/photo-tools.webp",
};

export const PRICE_TYPE: OptionType[] = [
  {
    option: "BTC",
    value: "1",
  },
  {
    option: "CNY",
    value: "2",
  },
  {
    option: "EUR",
    value: "3",
  },
  {
    option: "GBP",
    value: "4",
  },
  {
    option: "INR",
    value: "5",
  },
  {
    option: "USD",
    value: "6",
  },
];

export const PRICE_TAGS: OptionType[] = [
  {
    option: "Black Friday",
    value: "1",
  },
  {
    option: "Expired",
    value: "2",
  },
  {
    option: "Sale",
    value: "3",
  },
];

export const PRODUCT_CATEGORY: OptionType[] = [
  {
    option: "Clothing",
    value: "1",
  },
  {
    option: "Electronics",
    value: "2",
  },
  {
    option: "Furniture",
    value: "3",
  },
  {
    option: "Others",
    value: "4",
  },
  {
    option: "Real Estate",
    value: "5",
  },
];

export const OTHER_PRODUCTS_LENGTH = 4;

export const SOCIAL_PRODUCTS: TFieldSocial[] = [
  {
    id: "shopifyUrl",
    name: "shopifyUrl",
    label: "Shopify Handle",
    value: "",
  },
  {
    id: "facebookUrl",
    name: "facebookUrl",
    label: "Facebook Account",
    value: "",
  },
  {
    id: "instagramUrl",
    name: "instagramUrl",
    label: "Instagram Account",
    value: "",
  },
];
