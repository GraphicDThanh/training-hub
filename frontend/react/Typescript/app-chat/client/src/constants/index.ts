export const SOCKET_SERVER = 'localhost:8080';
export const DEFAULT_AVATAR_URL = 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQLfLjH2lLJ4LtZ6UmPxyt9I8a9N_q8q-JOqRVIee1pOAjO8gK9';
export const APP_TITLE = 'Graphic Chatroom';