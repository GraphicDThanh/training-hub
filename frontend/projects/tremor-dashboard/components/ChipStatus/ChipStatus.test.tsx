import { render } from "@testing-library/react";

// Constants
import { TRANSACTION_CLASS } from "@/constants";

// Components
import ChipStatus from "./ChipStatus";

describe("ChipStatus", () => {
  const { PENDING, INCREASE, DECREASE } = TRANSACTION_CLASS ?? {};
  const MOCK_VALUES = {
    PENDING: "Pending",
    INCREASE: "+",
    DECREASE: "-",
  };

  it("Match snapshot", () => {
    const { container } = render(
      <ChipStatus
        value={MOCK_VALUES.PENDING}
        extendedClass={PENDING.buttonClasses}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it("match snapshot with value is increase", () => {
    const { container } = render(
      <ChipStatus
        value={MOCK_VALUES.INCREASE}
        extendedClass={INCREASE.buttonClasses}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it("match snapshot with value is decrease", () => {
    const { container } = render(
      <ChipStatus
        value={MOCK_VALUES.DECREASE}
        extendedClass={DECREASE.buttonClasses}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
