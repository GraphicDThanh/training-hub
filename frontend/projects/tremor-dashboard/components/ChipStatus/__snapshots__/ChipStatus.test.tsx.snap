// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChipStatus Match snapshot 1`] = `
<div>
  <div
    class="tremor-Flex-root flex flex-row justify-center items-center shrink-0 w-9 h-9 border rounded-full text-primary dark:text-primary dark:hover:text-primary border-primary"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.005 3.1a1 1 0 1 1 1.99 0l-.388 6.35a.61.61 0 0 1-1.214 0L7.005 3.1ZM7 12a1 1 0 1 1 2 0 1 1 0 0 1-2 0Z"
      />
    </svg>
  </div>
</div>
`;

exports[`ChipStatus match snapshot with value is decrease 1`] = `
<div>
  <div
    class="tremor-Flex-root flex flex-row justify-center items-center shrink-0 w-9 h-9 border rounded-full text-attention dark:text-attention dark:hover:text-attention border-attention"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
      />
    </svg>
  </div>
</div>
`;

exports[`ChipStatus match snapshot with value is increase 1`] = `
<div>
  <div
    class="tremor-Flex-root flex flex-row justify-center items-center shrink-0 w-9 h-9 border rounded-full text-few dark:text-few dark:hove:text-few border-few"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m6.293 13.293 1.414 1.414L12 10.414l4.293 4.293 1.414-1.414L12 7.586z"
      />
    </svg>
  </div>
</div>
`;
